messages/src/did_doc/mod.rs

```diff
@@ -149,24 +149,18 @@ impl DidDoc {
         Ok(())
     }
 
-    pub fn recipient_keys(&self) -> Vec<String> {
+    pub fn recipient_keys(&self) -> MessagesResult<Vec<String>> {
         let service: AriesService = match self.service.get(0).cloned() {
             Some(service) => service,
-            None => return Vec::new(),
+            None => return Ok(Vec::new()),
         };
         let recipient_keys = service
             .recipient_keys
             .iter()
-            .filter_map(|recipient_key_entry| match self.find_key(recipient_key_entry) {
-                Ok(key) => Some(key.public_key_base_58),
-                Err(err) => {
-                    warn!(
-                        "ddo:: recipient_keys >> error occurred trying to recipient key entry {}, error: {}",
-                        recipient_key_entry, err
-                    );
-                    None
-                }
-            })
+            .map(|key_entry|
```

Comment:
Previously if there was invalid recipient key (neither naked key, neither referring a key in declared in diddoc), it would be ignored. Now we error on first processing error.